atc/engine/log_event_writer.go

```diff
@@ -94,11 +41,16 @@ func (writer *dbEventWriterWithSecretRedaction) Write(data []byte) (int, error)
 			return 0, nil
 		}
 		text = writer.dangling
+		writer.dangling = nil
 	}
 
 	payload := string(text)
 	if data != nil {
 		idx := strings.LastIndex(payload, "\n")
+		idxR := strings.LastIndex(payload, "\r")
+		if idx < idxR {
+			idx = idxR
+		}
```

Comment:
This should address #4799. Should see if there are some tests we can add though.